uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -326,6 +330,93 @@ private static function validateStartDateNotGreaterThanEndDate($params) {
         );
       }
   }
+
+  /**
+   * This method checks that the number of days for the Leave Request
+   * is not greater than the max_consecutive_leave_days for the absence type
+   *
+   * @param array $params
+   *   The params array received by the create method
+   *
+   * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   */
+  private static function validateLeaveDaysAgainstAbsenceTypeMaxConsecutiveLeaveDays($params) {
+    $absenceType = AbsenceType::findById($params['type_id']);
+    $fromDate = new DateTime($params['from_date']);
+    $toDate = new DateTime($params['to_date']);
+
+    $interval = $toDate->diff($fromDate);
+    $intervalInDays = $interval->format("%a");
+
+    if (!empty($absenceType->max_consecutive_leave_days) && $intervalInDays > $absenceType->max_consecutive_leave_days) {
+      throw new InvalidLeaveRequestException(
+        'Leave Request days cannot be greater than maximum consecutive days for absence type',
+        'leave_request_days_greater_than_max_consecutive_days',
+        'type_id'
+      );
+    }
+  }
+
+  /**
+   * This method checks if the absence type allows cancellation in advance of start date and that the leave request from_date
+   * should not be in the past in the event of a leave request cancellation by a user.
+   *
+   * Also checks that a user's leave request should not be cancelled if the absence type does not
+   * allow leave request cancellation
+   *
+   * @param array $params
+   *   The params array received by the create method
+   *
+   * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   */
+  private static function validateAbsenceTypeAllowRequestCancellationForLeaveRequestCancellation($params) {
+    $leaveRequestStatuses = array_flip(self::buildOptions('status_id'));
+    $leaveRequestIsForCurrentUser = CRM_Core_Session::getLoggedInContactID() == $params['contact_id'];
+    $isACancellationRequest = ($params['status_id'] == $leaveRequestStatuses['Cancelled']);
```

Comment:
Unfortunately, users can change the option values labels, so we should not rely on them in our code (its ok to use them on tests though). We need to use the name instead, which cannot be changed by the user. 

First, fetch the statuses with this:
```php
$leaveRequestStatuses = array_flip(self::buildOptions('status_id', 'validate'));
```

Next, to the check like this:
```php
$isACancellationRequest = ($params['status_id'] == $leaveRequestStatuses['cancelled']);
```